cutlery/_wordpiece.pyx

```diff
@@ -37,18 +42,61 @@ cdef class WordPieceProcessor:
             token_ids.append(match.piece_id)
             token_pieces.append(f"{prefix}{token[:match.prefix_len]}")
             token = token[match.prefix_len:]
-            pieces = &self.continuation_pieces
+            pieces = &self.continuation_piece_to_id
             prefix = "##"
         return token_ids, token_pieces
 
+    def decode(self, pieces: Iterable[int]) -> str:
+        """
+        Decode token piece identifiers into string. Raises a `ValueError` 
+        if any of the identifiers are invalid.
+
+            ids (Iterable[int]): Piece IDs.
+            RETURNS (str): Decoded string.
+        """
+        token_pieces = [self.lookup_piece_id(id)[0] for id in pieces]
+        return "".join(token_pieces)
+
     def get_initial(self, piece: str) -> int:
+        """
+        Returns the ID for the given initial piece. Raises a `KeyError` if 
+        the string isn't an initial piece.
+
+            piece (str): Initial piece string.
+            RETURNS (int): Piece ID.
+        """
         cdef unordered_map[string, size_t].const_iterator iter
-        iter = self.initial_pieces.const_find(piece.encode('utf8'))
-        if iter == self.initial_pieces.end():
-            raise KeyError(f"unknown initial piece {piece}")
+        iter = self.initial_piece_to_id.const_find(piece.encode('utf8'))
+        if iter == self.initial_piece_to_id.end():
+            raise KeyError(f"unknown initial piece `{piece}`")
         else:
             return deref(iter).second
 
+    def lookup_piece_id(self, piece: int) -> Tuple[str, bool]:
```

Comment:
I think I'd call this something like `id_to_piece`? (Or `get_piece`, but I think `id_to_piece` is a bit more explicit.) And then underscore the member variables.

Reply:
Done.